drivers/serial/pty.c

```diff
@@ -481,133 +491,34 @@ static ssize_t pty_read(FAR struct file *filep, FAR char *buffer, size_t len)
-           * source file as non-blocking, then (2) wait using poll() for the
-           * first byte to be received.  (3) Subsequent bytes would
-           * use file_read() without polling and would (4) terminate when no
-           * data is returned.
-           */
-
-          if (ntotal > 0)
-            {
-              int nsrc;
-
-              /* There are inherent race conditions in this test.  We lock
-               * the scheduler before the test and after the file_read()
-               * below to eliminate one race:  (a) We detect that there is
-               * data in the source file, (b) we are suspended and another
-               * thread reads the data, emptying the fifo, then (c) we
-               * resume and call file_read(), blocking indefinitely.
-               */
-
-              sched_lock();
-
-              /* Check how many bytes are waiting in the pipe */
-
-              ret = file_ioctl(&dev->pd_src, FIONREAD,
-                               (unsigned long)((uintptr_t)&nsrc));
-              if (ret < 0)
-                {
-                  sched_unlock();
-                  ntotal = ret;
-                  break;
-                }
-
-              /* Break out of the loop and return ntotal if the pipe is
-               * empty.  This is another race:  There fifo was empty when we
-               * called file_ioctl() above, but it might not be empty right
-               * now.  Losing that race should not lead to any bad behaviors,
-               * however, we the caller will get those bytes on the next
-               * read.
-               */
-
-              if (nsrc < 1)
-                {
-                  sched_unlock();
-                  break;
-                }
-
-              /* Read one byte from the source the byte.  This should not
-               * block.
-               */
-
-              nread = file_read(&dev->pd_src, &ch, 1);
-              sched_unlock();
-            }
-          else
-            {
-              /* Read one byte from the source the byte.  This call will
-               * block if the source pipe is empty.
-               *
-               * REVISIT: Should not block if the oflags include O_NONBLOCK.
-               * How would we ripple the O_NONBLOCK characteristic to the
-               * contained source pipe?  file_fcntl()?  Or FIONREAD? See the
-               * TODO comment at the top of this file.
-               */
-
-              nread = file_read(&dev->pd_src, &ch, 1);
-            }
-
-          /* Check if file_read was successful */
-
-          if (nread < 0)
-            {
-              ntotal = nread;
-              break;
-            }
-
           /* Perform input processing */
 
           /* \n -> \r or \r -> \n translation? */
 
-          if (ch == '\n' && (dev->pd_iflag & INLCR) != 0)
+          if (buffer[i] == '\n' && (dev->pd_iflag & INLCR) != 0)
             {
-               ch = '\r';
+               buffer[j++] = '\r';
```

Comment:
We change logic from original case. Now `'\r'` is not ignored anymore. Please confirm that this is intentional change and not a bug

Reply:
Ok, I keep the old logic completely.